src/errors.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export class CommitError extends Error {
+  cause: Error;
+  constructor(message: string, cause: Error) {
+    super(message);
+    this.cause = cause;
```

Comment:
The purpose of this is to wrap the original error, so someone catching it upstream can get at the original underlying issue?